### PR TITLE
Use gravatar for avatars

### PR DESCRIPTION
**Pull request by gela-d**

### CODE REVIEW

**Comment by gela-d** [comment]
@tavriaforever @tadatuta

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+modules.define('middleware__author', function (provide) {
```
Comment (tadatuta) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: is there any reason not to call it just `__gravatar`?

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    var crypto = require('crypto');
+
+    provide(function () {
```
Comment (tadatuta) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: why extra function wrapper here?

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        return function (req, res, next) {
+            var node = req.__data.node,
+                email = node.email[req.lang],
+                hash = crypto.createHash('md5').update(email, 'utf8').digest('hex');
```
Comment (tadatuta) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]:
``` js
if (!email) return next();
```

(there's no reason to generate hash if there's no email specified)

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+modules.define('middleware__author', function (provide) {
+
+    var crypto = require('crypto');
+
+    provide(function () {
+        return function (req, res, next) {
+            var node = req.__data.node,
```
Comment (tadatuta) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: you may cache `req.__data` into variable

File: src/blocks/server.blocks/middleware/middleware.node.js [hunk 5]
```diff
@@ -13,10 +13,11 @@ modules.define('middleware', [
         'middleware__page-menu',
         'middleware__lang-switcher',
         'middleware__page',
-        'middleware__error'
+        'middleware__error',
+        'middleware__author'
```
Comment (tadatuta) [inline src/blocks/server.blocks/middleware/middleware.node.js]: I'm not sure about order here

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js (hunk 1 above)
Comment (gela-d) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: @tadatuta of course not :)
Thank u, renamed

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js (hunk 2 above)
Comment (gela-d) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: @tadatuta I can't explain, but in out middlewares this is consistence

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js (hunk 3 above)
Comment (gela-d) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: @tadatuta sure, thank u

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js (hunk 4 above)
Comment (gela-d) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: @tadatuta update

File: src/blocks/server.blocks/middleware/middleware.node.js (hunk 5 above)
Comment (gela-d) [inline src/blocks/server.blocks/middleware/middleware.node.js]: @tavriaforever Can you review, do we need here this dependence?

File: src/blocks/server.blocks/middleware/__author/middleware__author.node.js (hunk 2 above)
Comment (gela-d) [inline src/blocks/server.blocks/middleware/__author/middleware__author.node.js]: @tavriaforever Can you tell?

**Comment by tavriaforever** [comment]
lgtm

**Pull request merged**